### ISSUE TITLE
Feature request: Dry Run

### ISSUE THREAD

**Issue by Germar** [bit-team/backintime#248]
It would be great to have a dry run option. It would help to see what files would be realy included. I'd prevent adding huge files by surprise like virtual machines images, iso and movies.

**Comment by bugspencor** [comment]
I'd like to see that too. It happens quite often that some big files end up unintentionally in the backup. Is there some other way (e.g. rsync commandline) to check in advance which new files would be backupped?

**Comment by colinl** [comment]
That would indeed be a very useful feature.

**Comment by ChrisK5** [comment]
Hallo,

yes, it would be nice to have such a feature implemented in Back In Time,
but just to let people like us know, people who searched for such an option, this feature is not necessarily needed.

You can perform a dry run by simply adding the option
--dry-run
 to
"Settings->Expert Options->Paste additional options to rsync"

This will perform the dry run we were looking for.
BackinTime / rsync will not copy any file, but maybe it will still create everything that it always creates, like the symlink and maybe the other data or files. Maybe the "old" symlink gets deleted because of that.
So if you perform a dry run like this you maybe back up the additional files that Back In Time creates beside the actual backup files.
I say maybe, because I do not know for sure how Back in Time will perform in the same Profil.
That's because I tested the dry run with an separate Profile to a separate folder, but with a backup of the additional files to restore them if needed, one should be fine, to make a dry run, i guess.

Kind regards
Chris

**Comment by emtiu** [comment]
Information from #249: a Dry Run would also help identify permission errors before starting a lengthy backup/transfer operation.

**Comment by buhtz** [comment]
Note: I got the hint that [LuckyBackup](https://sourceforge.net/projects/luckybackup/) (also using rsync) do have such a "simulation" feature.

**Comment by aryoda** [comment]
I think with `rsync --dry-run` plus a new BiT command line argument `--dry-run` plus a "Take a snapshot (dry run)" button this could be implemented quite easily.

I would use a signal file in the created snapshot folder (eg. `is_dry_run_result_only.txt`) to
1. capture and persist the dry run result log "as usual" without changing the whole back-end logic
2. allow the GUI to show a "dry run indicator" in the snapshots list
3. prevent restoring from a dry-run result
4. let the "auto-remove" feature ignore the dry run "snapshot" folder

Any more obstacles that we could face implementing this?

**Comment by buhtz** [comment]
Sounds like a f***ing awesome solution!

**Comment by buhtz** [comment]
Please note that the not-merged PR #2446 seems to be vibe-coded. Don't re-use it without validating it carefully.
